protocols/mdns/src/lib.rs

```diff
@@ -38,14 +38,26 @@
 use std::net::{Ipv4Addr, Ipv6Addr};
 use std::time::Duration;
 
+#[deprecated(
+    since = "0.42.0",
+    note = "Use re-exports that omit `Mdns` prefix, i.e. `libp2p::mdns::Config`"
+)]
+pub type MdnsConfig = Config;
+
+#[deprecated(
+    since = "0.42.0",
+    note = "Use re-exports that omit `Mdns` prefix, i.e. `libp2p::mdns::Event`"
+)]
+pub type MdnsEvent = Event;
+
 mod behaviour;
-pub use crate::behaviour::{GenMdns, MdnsEvent};
+pub use crate::behaviour::{Behaviour, Event};
 
 #[cfg(feature = "async-io")]
-pub use crate::behaviour::Mdns;
+pub use crate::behaviour::async_io;
 
 #[cfg(feature = "tokio")]
-pub use crate::behaviour::TokioMdns;
```

Comment:
If you'd add type aliases for this too then users would only be presented with deprecation warnings instead of compile errors.

Reply:
yup, makes sense. When putting the types was thinking that since this is a breaking change putting the types would be redundant, but if we also add for `Mdns` then it doesn't become a breaking change. Thanks Thomas :)